src/NHibernate/Persister/Entity/AbstractEntityPersister.cs

```diff
@@ -1722,13 +1722,13 @@ public object ForceVersionIncrement(object id, object currentVersion, ISessionIm
 		private SqlCommandInfo GenerateVersionIncrementUpdateString()
 		{
 			SqlUpdateBuilder update = new SqlUpdateBuilder(Factory.Dialect, Factory);
-			update.SetTableName(GetTableName(0));
+			update.SetTableName(RootTableName);
```

Comment:
There is a property, `VersionTableName`, which looks dedicated to this and is used in `GenerateSelectVersionString`. (This other method directly uses the `rootTableKeyColumnNames` field instead of `RootTableIdentifierColumnNames` which delegates to `RootTableKeyColumnNames`. All this seems a bit messy.)

Reply:
I've decided to obsolete `VersionedTableName` instead of using it.